File: src/main/java/com/github/solairerove/woodstock/controller/TaskController.java
//package com.github.solairerove.woodstock.controller;
//
//import com.github.solairerove.woodstock.dto.TaskDTO;
//import com.github.solairerove.woodstock.service.TaskService;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.http.HttpStatus;
//import org.springframework.http.MediaType;
//import org.springframework.http.ResponseEntity;
//import org.springframework.web.bind.annotation.PathVariable;
//import org.springframework.web.bind.annotation.RequestBody;
//import org.springframework.web.bind.annotation.RequestMapping;
//import org.springframework.web.bind.annotation.RequestMethod;
//import org.springframework.web.bind.annotation.RestController;
//
//@RestController
//@RequestMapping(value = "/api/modules/{moduleId}/tasks", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
//public class TaskController {
//
//    private final TaskService service;
//
//    @Autowired
//    public TaskController(TaskService service) {
//        this.service = service;
//    }
//
//    @RequestMapping(method = RequestMethod.POST)
//    public ResponseEntity create(@PathVariable Long moduleId, @RequestBody TaskDTO taskDTO) {
//        return new ResponseEntity<>(service.create(moduleId, taskDTO), HttpStatus.CREATED);
//    }
//
//    @RequestMapping(path = "/{taskId}")
//    public ResponseEntity get(@PathVariable Long moduleId, @PathVariable Long taskId) {
//        return new ResponseEntity<>(service.get(moduleId, taskId), HttpStatus.OK);
//    }
//
//    @RequestMapping
//    public ResponseEntity getAll(@PathVariable Long moduleId) {
//        return new ResponseEntity<>(service.getAll(moduleId), HttpStatus.OK);
//    }
//
//    @RequestMapping(path = "/{taskId}", method = RequestMethod.PUT)
//    public ResponseEntity update(@PathVariable Long moduleId, @PathVariable Long taskId, @RequestBody TaskDTO taskDTO) {
//        return new ResponseEntity<>(service.update(moduleId, taskId, taskDTO), HttpStatus.ACCEPTED);
//    }
//
//    @RequestMapping(path = "/{taskId}", method = RequestMethod.DELETE)
//    public ResponseEntity delete(@PathVariable Long moduleId, @PathVariable Long taskId) {
//        return new ResponseEntity<>(service.delete(moduleId, taskId), HttpStatus.ACCEPTED);
//    }
//}

File: src/main/webapp/app/components/task/task.controller.js
'use strict';

woodstock.controller('TaskCreateController', ['$scope', '$location', '$routeParams', 'TaskService',
    function ($scope, $location, $routeParams, TaskService) {
        $scope.task = new TaskService();

        $scope.addTask = function () {
            $scope.task.$create({moduleId: $routeParams.moduleId}, function () {
                $location.path('/modules/' + $routeParams.moduleId + '/tasks');
            });
        }
    }
]).controller('TaskListController', ['$scope', '$location', '$routeParams', 'TaskService',
    function ($scope, $location, $routeParams, TaskService) {
        TaskService.query({moduleId: $routeParams.moduleId}, function (data) {
            $scope.tasks = data;
            $scope.moduleId = $routeParams.moduleId;
        })
    }
]).controller('TaskDetailController', ['$scope', '$location', '$routeParams', 'TaskService',
    function ($scope, $location, $routeParams, TaskService) {
        TaskService.get({moduleId: $routeParams.moduleId, taskId: $routeParams.taskId}, function (data) {
            $scope.task = data;
            $scope.moduleId = $routeParams.moduleId;
        })
    }
]).controller('TaskEditController', ['$scope', '$location', '$routeParams', 'TaskService',
    function ($scope, $location, $routeParams, TaskService) {
        $scope.task = new TaskService();

        $scope.updateTask = function () {
            $scope.task.$update({moduleId: $routeParams.moduleId, taskId: $routeParams.taskId}, function () {
                $location.path('/modules/' + $routeParams.moduleId + '/tasks');
            });
        }
    }
]);
